### PR TITLE
[eclipse/xtext#1679] converted Xtend code to Java

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#1679] converted Xtend code to Java
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/formatting2/internal/GenericFormatterTestRequest.java [hunk 1]
```diff
@@ -10,24 +10,20 @@
...
-    this.formatter = formatter;
-  }
+	@Accessors
+	private GenericFormatter<? extends EObject> formatter;
+
+	public GenericFormatter<? extends EObject> getFormatter() {
+		return formatter;
+	}
+
+	public void setFormatter(final GenericFormatter<? extends EObject> formatter) {
```
Comment (cdietrich) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/formatting2/internal/GenericFormatterTestRequest.java:26]: final

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/formatting2/regionaccess/internal/RegionAccessTestHelper.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+	@Inject
+	private Provider<TextRegionAccessBuilder> textRegionAccessBuilder;
+
+	@Inject
+	private ValidationTestHelper validationTestHelper;
+
+	public ITextRegionAccess toTextRegionAccess(CharSequence file) throws Exception {
+		Root obj = parseHelper.parse(file.toString().replace("\r\n", "\n").trim());
+		this.validationTestHelper.assertNoErrors(obj);
```
Comment (cdietrich) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/formatting2/regionaccess/internal/RegionAccessTestHelper.java:49]: this.

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/generator/KeywordHelperTest.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+				IterableExtensions.sort(keywordHelper.getAllKeywords()).toString());
+		Assert.assertEquals("KW_EOF_1", keywordHelper.getRuleName("/EOF"));
+		Assert.assertEquals("KW_EOF", keywordHelper.getRuleName("EOF"));
+		Assert.assertEquals("Model", keywordHelper.getRuleName("model"));
+		Assert.assertEquals("AeOeUeaeOeUe", keywordHelper.getRuleName("�������"));
+	}
+
+	@Override
+	protected InputStream getAsStream(String model) {
+		return this.getAsStream(model, StandardCharsets.ISO_8859_1);
```
Comment (cdietrich) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/generator/KeywordHelperTest.java:72]: this.

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/formatting2/internal/GenericFormatter.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import org.eclipse.xtext.resource.XtextResource;
+
+/**
+ * @author Moritz Eysholdt - Initial contribution and API
+ */
+public abstract class GenericFormatter<T extends EObject> extends AbstractFormatter2 {
+	@SuppressWarnings("unchecked")
+	@Override
+	protected void _format(EObject obj, IFormattableDocument document) {
+		this.format(((T) obj), getRequest().getTextRegionAccess().getExtensions(), document);
```
Comment (cdietrich) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/formatting2/internal/GenericFormatter.java:24]: this.

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/filesystem/URIBasedFileSystemAccessTest.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+/**
+ * @author akos.kitta - Initial contribution and API
+ * 
+ * @see https://github.com/eclipse/xtext-core/issues/180
+ */
+@RunWith(XtextRunner.class)
+@InjectWith(XtextInjectorProvider.class)
+public class URIBasedFileSystemAccessTest {
+	private static String MISSING_RESOURCE_NAME = "someMissingResource";
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/filesystem/URIBasedFileSystemAccessTest.java:42]: Not final?

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/filesystem/URIBasedFileSystemAccessTest.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+	private static String EXISTING_RESOURCE_NAME = "someExistingResource";
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/filesystem/URIBasedFileSystemAccessTest.java:44]: Same here

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/filesystem/URIBasedFileSystemAccessTest.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+	@Inject
+	private URIBasedFileSystemAccess fsa;
+
+	@Inject
+	private IOutputConfigurationProvider configProvider;
+
+	private ExtensibleURIConverterImpl uriConverter = new ExtensibleURIConverterImpl();
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/filesystem/URIBasedFileSystemAccessTest.java:52]: Can be final

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/TypeReferenceTest.java [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl;
+import org.eclipse.xtext.xtext.generator.model.TypeReference;
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.google.common.collect.Lists;
+
+public class TypeReferenceTest {
+	/**
+	 * See https://bugs.eclipse.org/bugs/show_bug.cgi?id=480196
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/TypeReferenceTest.java:21]: This is fixed by now though

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/trace/node/GeneratorNodePerformanceTest.java [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+/**
+ * @author Dennis Huebner - Initial contribution and API
+ */
+public class GeneratorNodePerformanceTest {
+	private GeneratorNodeExtensions exts = new GeneratorNodeExtensions();
+
+//	private static final int numberOfLines = 800_000 // Before x sec - after 4.6 sec
+//	private static final int numberOfLines = 80_000 // Before 89 sec - after 0.43 sec
+	private static final int numberOfLines = 40_000; // Before 20 sec - after
+														// 0.32 sec
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/trace/node/GeneratorNodePerformanceTest.java:30]: Merge with comment above, it looks very weird floating by itself here

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/trace/node/GeneratorNodePerformanceTest.java [hunk 10]
```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2018 TypeFox (https://typefox.io) and others.
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/trace/node/GeneratorNodePerformanceTest.java:2]: Update copyright

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/trace/node/GeneratorNodePerformanceTest.java [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+					"indented2"));
+		}
+		exts.append(exts.appendNewLine(node), "dedented");
+		GeneratorNodeProcessor processor = new GeneratorNodeProcessor();
+		GeneratorNodeProcessor.Result result = processor.process(node);
+		String resultAsString = null;
+		if (result != null) {
+			resultAsString = result.toString();
+		}
+		Assert.assertFalse(StringExtensions.isNullOrEmpty(resultAsString));
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/trace/node/GeneratorNodePerformanceTest.java:53]: Just use `Strings.isNullOrEmpty`?

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/parser/fragments/FragmentExsPlainParsingTest.java [hunk 12]
```diff
@@ -14,12 +14,10 @@
 /**
  * @author Sebastian Zarnekow - Initial contribution and API
  */
-@SuppressWarnings("all")
 public class FragmentExsPlainParsingTest extends AbstractFragmentsPlainParsingTest {
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/parser/fragments/FragmentExsPlainParsingTest.java:17]: Copyright

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/trace/TraceRegionToStringTest.xtend [hunk 13]
```diff
@@ -8,15 +8,11 @@
  *******************************************************************************/
 package org.eclipse.xtext.generator.trace
 
-import java.util.List
-import org.eclipse.xtend.lib.annotations.Accessors
 import org.eclipse.xtext.generator.trace.TraceRegionToStringTester.Location
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/trace/TraceRegionToStringTest.xtend:11]: Copyright

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/formatting2/regionaccess/internal/RegionAccessCommentsTest.xtend [hunk 14]
```diff
@@ -24,7 +24,7 @@ import org.junit.runner.RunWith
 class RegionAccessCommentsTest {
 	@Inject extension RegionAccessTestHelper
 
-	@Test def void testDeleteFirstRegion1() {
+	@Test def void testDeleteFirstRegion1() throws Exception {
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/formatting2/regionaccess/internal/RegionAccessCommentsTest.xtend:27]: Copyright

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/formatting2/regionaccess/internal/RegionAccessDiffTest.xtend [hunk 15]
```diff
@@ -30,7 +30,7 @@ class RegionAccessDiffTest {
 	@Inject extension RegionAccessTestHelper
 	@Inject Serializer serializer
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/formatting2/regionaccess/internal/RegionAccessDiffTest.xtend:32]: Copyright

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/resource/containers/LiveShadowedChunkedContainerTest.xtend [hunk 16]
```diff
@@ -10,12 +10,8 @@ package org.eclipse.xtext.resource.containers
 
 import com.google.inject.Inject
 import com.google.inject.Provider
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/resource/containers/LiveShadowedChunkedContainerTest.xtend:12]: Copyright

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/resource/containers/ProjectConfig.java [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.xtext.util.UriUtil;
+import org.eclipse.xtext.workspace.IProjectConfig;
+import org.eclipse.xtext.workspace.ISourceFolder;
+import org.eclipse.xtext.workspace.IWorkspaceConfig;
+import org.eclipse.xtext.workspace.WorkspaceConfig;
+
+public class ProjectConfig implements IProjectConfig {
+	private URI path;
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/resource/containers/ProjectConfig.java:23]: final on all fields

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/resource/containers/SourceFolder.java [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+
+public class SourceFolder implements ISourceFolder {
+	private final URI path;
+
+	@Override
+	public String getName() {
+		return "src";
+	}
+
+	public SourceFolder(URI path) {
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/resource/containers/SourceFolder.java:22]: Put the constructor as first method?

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/generator/KeywordHelperTest.java [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+		super.setUp();
+		EPackage.Registry.INSTANCE.put(XMLTypePackage.eNS_URI, XMLTypePackage.eINSTANCE);
+		with(XtextStandaloneSetup.class);
+	}
+
+	@After
+	@Override
+	public void tearDown() throws Exception {
+		super.tearDown();
+		globalStateMemento.restoreGlobalState();
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/generator/KeywordHelperTest.java:48]: Hmm should we also undo this?
`EPackage.Registry.INSTANCE.put(XMLTypePackage.eNS_URI, XMLTypePackage.eINSTANCE);`

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/wizard/cli/CliProjectsCreatorMainTest.java [hunk 20]
```diff
@@ -0,0 +1,67 @@
...
+import org.eclipse.xtext.tests.TemporaryFolder;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+
+import com.google.common.collect.Lists;
+
+/**
+ * @author Karsten Thoms (karsten.thoms@itemis.de) - Initial contribution and
+ * API
```
Comment (tivervac) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/wizard/cli/CliProjectsCreatorMainTest.java:23]: Style: move this up to the line above

File: org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/RuntimeTestProjectDescriptor.java [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+	public boolean isPartOfGradleBuild() {
+		return getConfig().getPreferredBuildSystem() == BuildSystem.GRADLE;
+	}
+
+	@Override
+	public boolean isPartOfMavenBuild() {
+		return getConfig().getPreferredBuildSystem() == BuildSystem.MAVEN;
+	}
+
+	public RuntimeTestProjectDescriptor(TestedProjectDescriptor testedProject) {
```
Comment (tivervac) [inline org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/RuntimeTestProjectDescriptor.java:38]: I would put this as the first method

File: org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/UiTestProjectDescriptor.java [hunk 22]
```diff
@@ -0,0 +1,56 @@
...
+	public boolean isPartOfMavenBuild() {
+		return true;
+	}
+
+	@Override
+	public boolean needsUiHarness() {
+		return true;
+	}
+
+	public UiTestProjectDescriptor(TestedProjectDescriptor testedProject) {
```
Comment (tivervac) [inline org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/UiTestProjectDescriptor.java:53]: I'd put this as the first method

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/TypeReferenceTest.java (hunk 8 above)
Comment (cdietrich) [reply org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/TypeReferenceTest.java:21]: did not want to look into the details thus is just took over the comment

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/trace/node/GeneratorNodePerformanceTest.java (hunk 9 above)
Comment (cdietrich) [reply org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/trace/node/GeneratorNodePerformanceTest.java:30]: autoformatter is arggg

Comment (tivervac) [reply org.eclipse.xtext.tests/src/org/eclipse/xtext/generator/trace/node/GeneratorNodePerformanceTest.java:30]: Yeah, for some reason the default formatting length for comments is 80 chars. We've changed it to 120 in our codebase, same as for java lines.
Not sure why you'd want a different allowed length for your comments than for your code

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/generator/KeywordHelperTest.java (hunk 19 above)
Comment (cdietrich) [reply org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/generator/KeywordHelperTest.java:48]: restoreGlobalState

**Pull request merged**